routingv8/client.go

```diff
@@ -0,0 +1,149 @@
+package routingv8
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+const (
+	userAgent = "einride/here-go"
+)
+
+// RouteService handles communication with the route-related methods of the v7 HERE API.
+type RouteService service
+
+// MatrixService handles communication with the matrix-related methods of the v7 HERE API.
+type MatrixService service
+
+type Client struct {
+	// HTTP client used to communicate with the API.
+	client *http.Client
+
+	UserAgent string
+
+	// Reuse a single struct instead of allocating one for each service on the heap.
+	common service
```

Comment:
the naming here is a bit unintuitive - what is it really?

Reply:
Copy-paste from v7 🙈  Removed it as we're not using it in v8.